diff --git a/app/src/main/java/org/simple/clinic/search/PatientSearchScreen.kt b/app/src/main/java/org/simple/clinic/search/PatientSearchScreen.kt
index bc724b51c..7c67e1085 100644
--- a/app/src/main/java/org/simple/clinic/search/PatientSearchScreen.kt
+++ b/app/src/main/java/org/simple/clinic/search/PatientSearchScreen.kt
@@ -16,7 +16,9 @@ import org.simple.clinic.allpatientsinfacility.migration.AllPatientsInFacilityVi
 import org.simple.clinic.allpatientsinfacility.migration.ExposesUiEvents
 import org.simple.clinic.allpatientsinfacility_old.AllPatientsInFacilityListScrolled
 import org.simple.clinic.allpatientsinfacility_old.AllPatientsInFacilitySearchResultClicked
+import org.simple.clinic.analytics.Analytics
 import org.simple.clinic.bindUiToController
+import org.simple.clinic.mobius.migration.Architecture
 import org.simple.clinic.mobius.migration.MobiusMigrationConfig
 import org.simple.clinic.mobius.migration.RevertToCommit
 import org.simple.clinic.patient.PatientSearchCriteria
@@ -31,7 +33,6 @@ import org.simple.clinic.widgets.UiEvent
 import org.simple.clinic.widgets.hideKeyboard
 import org.simple.clinic.widgets.showKeyboard
 import org.threeten.bp.Instant
-import timber.log.Timber
 import java.util.UUID
 import javax.inject.Inject
 
@@ -60,13 +61,15 @@ class PatientSearchScreen(context: Context, attrs: AttributeSet) : RelativeLayou
   lateinit var mobiusMigrationConfig: MobiusMigrationConfig
 
   private val allPatientsInFacilityView by unsafeLazy {
-    allPatientsViewStub.layoutResource = if (mobiusMigrationConfig.useAllPatientsInFacilityView) {
-      Timber.d("Using AllPatientsInFacilityView (Mobius)")
-      R.layout.view_allpatientsinfacility
+    val (layoutResId, architecture) = if (mobiusMigrationConfig.useAllPatientsInFacilityView) {
+      R.layout.view_allpatientsinfacility to Architecture.MOBIUS
     } else {
-      Timber.d("Using AllPatientsInFacilityView (OG Architecture)")
-      R.layout.view_allpatientsinfacility_old
+      R.layout.view_allpatientsinfacility_old to Architecture.ORIGINAL
     }
+
+    Analytics.reportArchitectureMigration("AllPatientsInFacilityView", architecture, "PatientSearchScreen")
+
+    allPatientsViewStub.layoutResource = layoutResId
     allPatientsViewStub.inflate() as ExposesUiEvents
   }
 
