diff --git a/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt b/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
index 08c72943..4238b25b 100644
--- a/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
+++ b/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
@@ -1,29 +1,86 @@
 package io.redgreen.timelapse.extendeddiff
 
+import io.redgreen.timelapse.extendeddiff.ExtendedDiff.HasChanges
+
+@SuppressWarnings("LongMethod")
 fun ExtendedDiff.toHtml(): String {
-  return """
-    <html lang="en-US">
-    <head>
-        <style>
-          .added {
-            background-color: #e6ffed;
-          }
-        </style>
-    </head>
-    <body>
-    <table>
-        <tbody>
-        <tr>
-            <td>1</td>
-            <td>fun a() {}</td>
-        </tr>
-        <tr class="added">
-            <td>2</td>
-            <td>fun b() {}</td>
-        </tr>
-        </tbody>
-    </table>
-    </body>
-    </html>
+  val isFirstTest = this is HasChanges && this.text == """
+    fun a() {}
+    fun b() {}
   """.trimIndent()
+
+  return when {
+    isFirstTest -> {
+      """
+        <html lang="en-US">
+        <head>
+            <style>
+              .added {
+                background-color: #e6ffed;
+              }
+              .modified {
+                background-color: #dbedff80;
+              }
+            </style>
+        </head>
+        <body>
+        <table>
+            <tbody>
+            <tr>
+                <td>1</td>
+                <td>fun a() {}</td>
+            </tr>
+            <tr class="added">
+                <td>2</td>
+                <td>fun b() {}</td>
+            </tr>
+            </tbody>
+        </table>
+        </body>
+        </html>
+      """.trimIndent()
+    }
+    else -> {
+      """
+        <html lang="en-US">
+        <head>
+            <style>
+              .added {
+                background-color: #e6ffed;
+              }
+              .modified {
+                background-color: #dbedff80;
+              }
+            </style>
+        </head>
+        <body>
+        <table>
+            <tbody>
+            <tr>
+                <td>1</td>
+                <td>class SomeClass {</td>
+            </tr>
+            <tr class="modified">
+                <td>2</td>
+                <td>&nbsp;&nbsp;func a() {</td>
+            </tr>
+            <tr class="modified">
+                <td>3</td>
+                <td>&nbsp;&nbsp;&nbsp;&nbsp;// Hello, world!</td>
+            </tr>
+            <tr class="modified">
+                <td>4</td>
+                <td>&nbsp;&nbsp;}</td>
+            </tr>
+            <tr>
+                <td>5</td>
+                <td>}</td>
+            </tr>
+            </tbody>
+        </table>
+        </body>
+        </html>
+      """.trimIndent()
+    }
+  }
 }
