diff --git a/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt b/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
index 52fc6a27..3cc78dbb 100644
--- a/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
+++ b/app/src/main/kotlin/io/redgreen/timelapse/extendeddiff/ExtendedDiffHtml.kt
@@ -95,6 +95,9 @@ fun ExtendedDiff.toHtml(): String {
               .modified {
                 background-color: #dbedff80;
               }
+              .deleted {
+                background-color: #ffdce0;
+              }
             </style>
         </head>
         <body>
